sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -303,10 +316,17 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
         this.width = width
         this.height = height
         shouldExit = false
-        eventQueue.clear()
-        renderEventQueue.clear()
+        // we clean only Mapbox events to avoid outdated runnables associated with previous EGL context
+        val iterator = renderEventQueue.iterator()
+        while (iterator.hasNext()) {
+          val next = iterator.next()
+          if (next.eventType == EventType.MAPBOX) {
+            iterator.remove()
+          }
+        }
         snapshotQueue.clear()
```

Comment:
What about snapshots though? Should we clear them (e.g. if snapshot was called prior to onSurfaceCreated)

Reply:
imo yes
making a snapshot looks to me as an atomic operation - user requests current state of the map. If there's not map atm (surface not created) - we should return null. I'll check this case though, it might not work as expected now I guess (such runnables could be just silently discarded)

Reply:
refactored snapshot runnables to be actual render runnables